### ISSUE TITLE
Applying change to yText with applyDelta does not end in same state as using Delta directly

### ISSUE THREAD

**Issue by deeg** [yjs/yjs#503]
> Please save me some time and use the following template. In 90% of all issues I can't reproduce the problem because I don't know what exactly you are doing, in which environment, or which y-* version is responsible. Just use the following template even if you think the problem is obvious.

**Checklist**

* [X] Are you reporting a bug? Use github issues for bug reports and feature requests. For general questions, please use https://discuss.yjs.dev/
* [X] Try to report your issue in the correct repository. Yjs consists of many modules. When in doubt, report it to https://github.com/yjs/yjs/issues/

**Describe the bug**
Applying a Delta change event to yText does not end in the same end state as applying that same change using Delta directly.

`new Delta(INITIAL_CONTENT).compose(CHANGE_EVENT)`

does not equal

```
var yDoc = new Y.Doc();
var yText = yDoc.getText('id');
ytext.applyDelta(INITIAL_CONTENT)
ytext.applyDelta(CHANGE_EVENT)
ytext.toDelta()
```

![CleanShot 2023-02-17 at 11 53 00](https://user-images.githubusercontent.com/2857769/219777789-2da252ae-308a-4c3d-bec9-1c770799f8bf.png)

**To Reproduce**
Steps to reproduce the behavior please see the following stack blitz:

https://stackblitz.com/edit/typescript-ddesey?file=index.ts

**Expected behavior**
I expect the end state using Delta only to be the same as applying that change with yText.applyDelta

**Environment Information**
YJS: 13.5.46

**Comment by ralphiee22** [comment]
Just want to add another example to this issue, where attributes are not properly removed from YJS (when using `null` delta syntax).

https://stackblitz.com/edit/typescript-u9rykh?file=index.ts

**Comment by dmonad** [comment]
Thanks @ralphiee22 ! 

Just fixed both issues in Yjs@13.5.48. Next step I'll add some more tests similar to the ones that you created.
